pkg/cmd/cli/cmd/set/probe.go

```diff
@@ -0,0 +1,409 @@
+  $ %[1]s probe dc/router --readiness --get-url=https://127.0.0.1:1936/stats
+
+  # Set only the initial-delay-seconds field on all deployments
+  $ %[1]s probe dc --all --readiness --initial-delay-seconds=30`
+)
+
+type ProbeOptions struct {
+	Out io.Writer
+	Err io.Writer
+
+	Filenames         []string
+	ContainerSelector string
+	Selector          string
+	All               bool
+
+	Builder *resource.Builder
+	Infos   []*resource.Info
+
+	Encoder runtime.Encoder
+
+	ShortOutput bool
+	Mapper      meta.RESTMapper
+
+	PrintObject            func(runtime.Object) error
+	UpdatePodSpecForObject func(runtime.Object, func(spec *kapi.PodSpec) error) (bool, error)
+
+	Readiness bool
+	Liveness  bool
+	Remove    bool
+
+	OpenTCPSocket string
+	HTTPGet       string
+	Command       []string
+
+	FlagSet       func(string) bool
+	HTTPGetAction *kapi.HTTPGetAction
+
+	// Length of time before health checking is activated.  In seconds.
+	InitialDelaySeconds *int
+	// Length of time before health checking times out.  In seconds.
+	TimeoutSeconds *int
+	// How often (in seconds) to perform the probe.
+	PeriodSeconds *int
+	// Minimum consecutive successes for the probe to be considered successful after having failed.
+	// Must be 1 for liveness.
+	SuccessThreshold *int
+	// Minimum consecutive failures for the probe to be considered failed after having succeeded.
+	FailureThreshold *int
+}
+
+// NewCmdProbe implements the set probe command
+func NewCmdProbe(fullName string, f *clientcmd.Factory, out, errOut io.Writer) *cobra.Command {
+	options := &ProbeOptions{
+		Out: out,
+		Err: errOut,
+
+		ContainerSelector: "*",
+	}
+	cmd := &cobra.Command{
+		Use:     "probe RESOURCE/NAME --readiness|--liveness (--get-url=URL|--open-tcp=PORT|-- CMD)",
+		Short:   "Update a probe on a pod template",
+		Long:    probeLong,
+		Example: fmt.Sprintf(probeExample, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(options.Complete(f, cmd, args))
+			kcmdutil.CheckErr(options.Validate())
+			if err := options.Run(); err != nil {
+				// TODO: move met to kcmdutil
+				if err == cmdutil.ErrExit {
+					os.Exit(1)
+				}
+				kcmdutil.CheckErr(err)
+			}
+		},
+	}
+
+	kcmdutil.AddPrinterFlags(cmd)
+	cmd.Flags().StringVarP(&options.ContainerSelector, "containers", "c", options.ContainerSelector, "The names of containers in the selected pod templates to change - may use wildcards")
+	cmd.Flags().StringVarP(&options.Selector, "selector", "l", options.Selector, "Selector (label query) to filter on")
+	cmd.Flags().BoolVar(&options.All, "all", options.All, "Select all resources in the namespace of the specified resource types")
+	cmd.Flags().StringSliceVarP(&options.Filenames, "filename", "f", options.Filenames, "Filename, directory, or URL to file to use to edit the resource.")
+
+	cmd.Flags().BoolVar(&options.Remove, "remove", options.Remove, "If true, remove the specified probe(s).")
```

Comment:
There are only two, so it didn't seem like we needed it.  Triggers are a
list, this is more of a named struct.

On Sun, Feb 21, 2016 at 7:59 PM, Fabiano Franz <notifications@github.com>
wrote:

> In pkg/cmd/cli/cmd/set/probe.go
> <https://github.com/openshift/origin/pull/7331#discussion_r53578238>:
>
> > +				// TODO: move met to kcmdutil
> > +				if err == cmdutil.ErrExit {
> > +					os.Exit(1)
> > +				}
> > +				kcmdutil.CheckErr(err)
> > +			}
> > +		},
> > +	}
> > +
> > +	kcmdutil.AddPrinterFlags(cmd)
> > +	cmd.Flags().StringVarP(&options.ContainerSelector, "containers", "c", options.ContainerSelector, "The names of containers in the selected pod templates to change - may use wildcards")
> > +	cmd.Flags().StringVarP(&options.Selector, "selector", "l", options.Selector, "Selector (label query) to filter on")
> > +	cmd.Flags().BoolVar(&options.All, "all", options.All, "Select all resources in the namespace of the specified resource types")
> > +	cmd.Flags().StringSliceVarP(&options.Filenames, "filename", "f", options.Filenames, "Filename, directory, or URL to file to use to edit the resource.")
> > +
> > +	cmd.Flags().BoolVar(&options.Remove, "remove", options.Remove, "If true, remove the specified probe(s).")
>
> Add --remove-all like in triggers?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/7331/files#r53578238>.
>